it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java

```diff
@@ -1,111 +1,70 @@
 
     @Rule
     public TestName test = new TestName();
 
-    public String getTestName() {
-        return test.getMethodName();
-    }
-
-    public LoggingParams getDockerLoggingParams() {
-        return LoggingParams.builder()
-                .testName(getTestName())
-                .enabled(false)
-                .build();
-    }
-
-    public Scenario getVmScenario(int numNodes) {
-        VmUniverseFixture universeFixture = new VmUniverseFixture();
-        universeFixture.setNumNodes(numNodes);
-
-        VmUniverseParams universeParams = universeFixture.data();
+    private UniverseManager universeManager;
 
-        ApplianceManager manager = ApplianceManager.builder()
-                .universeParams(universeParams)
+    @Before
+    public void setUp() {
+        universeManager = UniverseManager.builder()
+                .testName(test.getMethodName())
+                .universeMode(UniverseMode.DOCKER)
+                .corfuServerVersion(APP_UTIL.getAppVersion())
                 .build();
-
-        //Assign universe variable before deploy prevents resources leaks
-        universe = UNIVERSE_FACTORY.buildVmUniverse(universeParams, manager);
-        universe.deploy();
-
-        return Scenario.with(universeFixture);
     }
 
-    public Scenario getDockerScenario(int numNodes, Set<Integer> metricsPorts) {
-        UniverseFixture universeFixture = new UniverseFixture();
-        universeFixture.setNumNodes(numNodes);
-        universeFixture.setMetricsPorts(metricsPorts);
+    public <T extends Fixture<UniverseParams>> UniverseWorkflow workflow(
+            Consumer<UniverseWorkflow<T>> action) {
 
-        //Assign universe variable before deploy prevents resources leaks
-        universe = UNIVERSE_FACTORY.buildDockerUniverse(universeFixture.data(), docker, getDockerLoggingParams());
-        universe.deploy();
-
-        return Scenario.with(universeFixture);
-    }
-
-    public Scenario<UniverseParams, AbstractUniverseFixture<UniverseParams>> getScenario() {
-        final int defaultNumNodes = 3;
-        return getScenario(defaultNumNodes, ImmutableSet.of());
+        return universeManager.workflow(action);
     }
 
-    public Scenario<UniverseParams, AbstractUniverseFixture<UniverseParams>> getScenario(int numNodes) {
-        return getScenario(numNodes, ImmutableSet.of());
-    }
+    private static class AppUtil {
+        private static final String POM_FILE = "pom.xml";
+
+        /**
+         * Provides a current version of this project. It parses the version from pom.xml
+         *
+         * @return maven/project version
+         */
+        public String getAppVersion() {
+            String version = System.getProperty("project.version");
+            if (version != null && !version.isEmpty()) {
+                return version;
+            }
+
+            return parseAppVersionInPom();
+        }
 
-    public Scenario<UniverseParams, AbstractUniverseFixture<UniverseParams>>getScenario(
```

Comment:
How are the metrics port suppose to be passed? Right now this breaks all tests that rely on this feature. Here it's how it is being initialized right now:

```
        getScenario(3, Collections.singleton(metricsPort)).describe((fixture, testCase) -> {
            CorfuCluster corfuCluster = universe.getGroup(fixture.getCorfuCluster().getName());
            CorfuClient corfuClient = corfuCluster.getLocalCorfuClient(metricsPort);
```
What is the new workflow suppose to be?

Reply:
```
wf.setupDocker(fixture -> {
                //fixture.getSupportServer(). define whatever you need
                fixture.getSupportServer().metricPorts(metricPorts);
            });
...
```